### PR TITLE
fix reticket flow to handle noncustodial wallets

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Reticket doesn't work when we don't have control of the private
key. This change makes the reticket flow use the signTransaction
and sendSignedTransaction helpers, which have special cases for
handling wallets where we don't have custody of the private key.

Also disables gas tank for metamask users.
Addresses #293 #353

### CODE REVIEW

File: src/lib/reticket.js [hunk 1]
```diff
@@ -143,26 +148,44 @@ export async function reticketPointBetweenWallets({
...
-    return {
-      raw: hexify(stx.serialize()),
-      signed: stx,
-    };
-  });
+  console.log(txs);
+
+  let txPairs = await Promise.all(
+    txs.map(async (tx, idx) => {
+      const stx = await signTransaction({
```
Comment (Fang-) [inline src/lib/reticket.js]: I suppose the UX for this in the previously-not-supported cases is just repeated pop-ups/prompts to do the signing?

**Comment by vvisigoth** [comment]
I just tested with Ledger and doestn' work, I've attached errors. @liam-fitzgerald do you have a hardware wallet to test this with? If not, email me.

Screen Shot 2020-01-14 at 4 16 23 PM

Screen Shot 2020-01-14 at 4 16 49 PM

**Comment by Fang-** [comment]
@vvisigoth Just in case, can you confirm you're running on latest Ledger firmware? #364 updated the dependencies for Ledger, that may or may not break older firmwares.

**Comment by vvisigoth** [comment]
OK, updated firmware and now I have different problems.

Screen Shot 2020-01-14 at 4 37 52 PM

Screen Shot 2020-01-14 at 4 38 01 PM

**Comment by vvisigoth** [comment]
Talked to @Fang- offline. He's double checking we have correct firmware support.

**Comment by vvisigoth** [comment]
OK, testing on Ledger. I fixed some of these issues by using the latest firmware here https://github.com/urbit/bridge/pull/372 . Still running into the "U2_LEDGER DEVICE INELLIGIBLE" error. The error seems to pop up after I've approved the "Fields" of the tx on the ledger device, but before I've been asked to "Approve and Send" the first transaction.

Other transactinos still work, just falling over on the Reticket flow.

File: src/lib/reticket.js (hunk 1 above)
Comment (Fang-) [reply src/lib/reticket.js]: Perhaps this is too eager in sending transactions to the device, and we want to actually run this in sequence? ie, maybe not rely on `.map` here? Not sure, but seems like something worth trying if we don't have any other leads.

Comment (liam-fitzgerald) [reply src/lib/reticket.js]: thanks for the heads up. yeah the popups/prompts just repeat

**Comment by liam-fitzgerald** [comment]
Latest commit works with my ledger, would you be able to source a trezor for testing?

**Comment by Fang-** [comment]
@vvisigoth I won't have access to a Trezor (or any hw wallet) until next week. Can you give this a test run?

**Comment by liam-fitzgerald** [comment]
ping @Fang- @vvisigoth do either of you have access to a trezor yet?

**Comment by Fang-** [comment]
Just tested with a Trezor. Doing this from a just-reloaded Bridge page works fine. When I tried doing this directly after accepting an incoming transfer, I got the following error:

image

But all those transactions still succeeded, so not sure what was up here. This also seems like a more general issue (potentially stale nonces) rather than a flow-specific one, and/or might have been present in the first place.

Merging this as confirmed fixed!

**Pull request merged**